### ISSUE TITLE
`naming-convention` rule fails when querying `__typename`

### ISSUE THREAD

**Issue by rintaun** [dotansimha/graphql-eslint#664]
**Describe the bug**

The `__typename` meta-field is provided by the GraphQL specification as a way to introspect the current type at any point in a query. This is useful, for example, when using object spreads, e.g.:

```graphql
mutation DoSomething {
  doSomething(input: ...) {
    userErrors {
      ... on Error {
        message
        path
        __typename
      }
    }
    ...
}
```

In the above example, `__typename` allows the client to know the specific type of errors that occurred.

There is currently no way to allow this without setting `leadingUnderscore` to `allow`, which would also allow any other underscore-preceded field name.

**To Reproduce**

```
git clone git@github.com:rintaun/graphql-eslint-naming-convention.git
npm install
npm run lint
```

**Expected behavior**

Linting passes, perhaps by one of the following mechanisms:

1. just ignoring `__typename`
2. having a configuration option to add exceptions for particular field names, i.e. `__typename`
3. ignoring anything beginning with `__`, as that prefix is reserved by GraphQL introspection

I think that option 3 would likely be simplest to implement and use.

**Environment:**

- OS: Ubuntu 18.04.2
- `@graphql-eslint/...`: 2.2.0
- NodeJS: 14.4.0 / 16.10.0

**Additional context**

N/A

**Comment by B2o5T** [comment]
Hi @rintaun 👍, thanks for the report! Nice catch

You are right, @dotansimha I think we need to ignore `SelectionSet` fields, and only check for trailing/leading underscore if there is an `alias` renaming. I added a fix. We can fix this on the next major release with new options API

**Comment by dotansimha** [comment]
Fixed in `@graphql-eslint/eslint-plugin@3.0.0`
